datafusion/core/src/physical_optimizer/enforce_sorting.rs

```diff
@@ -775,43 +776,39 @@ mod tests {
     #[tokio::test]
     async fn test_remove_unnecessary_sort_window_multilayer() -> Result<()> {
         let schema = create_test_schema()?;
-        let source = memory_exec(&schema);
 
         let sort_exprs = vec![sort_expr_options(
             "non_nullable_col",
-            &source.schema(),
+            &schema,
             SortOptions {
                 descending: true,
                 nulls_first: true,
             },
         )];
-        let sort = sort_exec(sort_exprs.clone(), source);
-        // Add dummy layer propagating Sort above, to test whether sort can be removed from multi layer before
-        let coalesce_batches = coalesce_batches_exec(sort);
 
-        let window_agg =
-            bounded_window_exec("non_nullable_col", sort_exprs, coalesce_batches);
+        let builder = TestPlanBuilder::new_memory_exec(&schema)
```

Comment:
This shows a good example of the difference -- the plan gets built in builder chain rather than several individual function calls